src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -333,7 +389,20 @@ public void SetTitle(string title)
 
         public virtual void Show()
         {
-            SetWindowLongPtr(_hwnd, (int)WindowLongParam.GWL_HWNDPARENT, IntPtr.Zero);
+            IntPtr ownerHandle = IntPtr.Zero;
+
+            // Window was potentially shown as dialog so we have to restore owner.
+            if (_ownerWindow != null)
```

Comment:
Is this a supported scenario to call `Show -> Close -> Show` multiple times? Once I close a window this is final in my understanding. For this use-case someone would call `Show -> Hide -> Activate`

Reply:
`Close` is final, but user can call `Show -> Hide -> Show` or even `Show -> Hide -> ShowDialog`. WPF seems to support this as well: https://docs.microsoft.com/en-us/dotnet/api/system.windows.window.hide?view=netframework-4.8.